recompose-idea/src/main/kotlin/recompose/plugin/copypaste/RecomposeCopyPasteProcessor.kt

```diff
@@ -58,14 +59,28 @@ class RecomposeCopyPasteProcessor : CopyPastePostProcessor<TextBlockTransferable
     override fun extractTransferableData(
         content: Transferable
     ): List<TextBlockTransferableData> {
-        if (content.isDataFlavorSupported(CopiedXMLCode.DATA_FLAVOR)) {
-            // There's some matching data in this paste, let's return it to process it.
-            return listOf(
-                content.getTransferData(CopiedXMLCode.DATA_FLAVOR) as TextBlockTransferableData
-            )
-        }
 
-        return emptyList()
+        return if (content.isDataFlavorSupported(DataFlavor.stringFlavor)) {
+            val text = content.getTransferData(DataFlavor.stringFlavor) as String
+            val xmlPreamble = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+
+            when {
+                content.isDataFlavorSupported(CopiedXMLCode.DATA_FLAVOR) -> {
+                    // There's some matching data in this paste, let's return it to process it.
+                    listOf(
+                        content.getTransferData(CopiedXMLCode.DATA_FLAVOR) as TextBlockTransferableData
+                    )
+                }
+                text.contains(xmlPreamble, true) -> {
+                    listOf(CopiedXMLCode(text, intArrayOf(0), intArrayOf(0)) as TextBlockTransferableData)
```

Comment:
Shouldn't we set the `endOffset` to the last index of `text`? Right now the offsets are not used, but eventually they will, I guess (#5).

Reply:
To be honest, i wasn't sure what the correct values for the offsets are. I looked at what offsets are used in collectTransferableData() when i copy the text inside Intellij and they we both 0.

Reply:
Oh, I see. Then this is something to look at when tackling #5. I was expecting the offsets to mark the selection area inside the text, since right now that always contains the whole document. But if it's always 0 right now then let's just use 0 here for now too. 👍